features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/BitmapPoolTest.kt

```diff
@@ -0,0 +1,216 @@
+
+        val width = forge.anInt(1, 200)
+        val height = forge.anInt(1, 200)
+        val allocSize = forge.anInt(100, 10000)
+
+        whenever(mockBitmap.config).thenReturn(mockConfig)
+        whenever(mockSecondBitmap.config).thenReturn(mockConfig)
+        whenever(mockBitmap.width).thenReturn(width)
+        whenever(mockBitmap.height).thenReturn(height)
+        whenever(mockBitmap.allocationByteCount).thenReturn(allocSize)
+        whenever(mockSecondBitmap.width).thenReturn(width)
+        whenever(mockSecondBitmap.height).thenReturn(height)
+        whenever(mockSecondBitmap.allocationByteCount).thenReturn(allocSize)
+    }
+
+    @Test
+    fun `M return null W get() { cache has no bitmaps of the right size }`() {
+        // Given
+        val width = mockBitmap.width
+        val height = mockBitmap.height
+        val config = mockConfig
+
+        // When
+        val bitmap = testedCache.getBitmapByProperties(width, height, config)
+
+        // Then
+        assertThat(bitmap).isNull()
+    }
+
+    @Test
+    fun `M return a bitmap W get() { free bitmap in cache }`() {
+        // Given
+        val width = mockBitmap.width
+        val height = mockBitmap.height
+        val config = mockConfig
+        val key = "$width-$height-$config"
+        whenever(mockLruCache.get(key)).thenReturn(mockBitmap)
+        val bitmapsBySize = HashMap<String, HashSet<Bitmap>>()
+        bitmapsBySize.put(key, hashSetOf(mockBitmap))
+        testedCache.setBitmapsBySize(bitmapsBySize)
+
+        // When
+        val cacheItem = testedCache.getBitmapByProperties(width, height, config)
+
+        // Then
+        assertThat(cacheItem).isEqualTo(mockBitmap)
+    }
+
+    @Test
+    fun `M return null W get() { bitmap in cache but it's already in use }`() {
+        // Given
+        val width = mockBitmap.width
+        val height = mockBitmap.height
+        val config = mockConfig
+        val key = "$width-$height-$config"
+        whenever(mockLruCache.get(key)).thenReturn(mockBitmap)
+        val bitmapsBySize = HashMap<String, HashSet<Bitmap>>()
+        bitmapsBySize.put(key, hashSetOf(mockBitmap))
+        testedCache.setBitmapsBySize(bitmapsBySize)
+        testedCache.setUsedBitmaps(hashSetOf(mockBitmap))
+
+        // When
+        val cacheItem = testedCache.getBitmapByProperties(width, height, config)
+
+        // Then
+        Assertions.assertThat(cacheItem).isNull()
+    }
+
+    @Test
+    fun `M call LruCache put with index 1 W put() { first bitmap }`() {
+        // Given
+        val width = mockBitmap.width
+        val height = mockBitmap.height
+        val config = mockConfig
+        val key = "$width-$height-$config"
+
+        // When
+        testedCache.put(mockBitmap)
+
+        // Then
+        verify(mockLruCache).put(key + "-${testedCache.bitmapIndex}", mockBitmap)
+    }
+
+    @Test
+    fun `M call LruCache put with index 2 W put() { bitmapsBySize has an item }`() {
```

Comment:
you should not test these methods, they are part of your internal logic and here you need to test the public API. Instead here you could have tested that if you `put` 2 values in the pool and you call `get` 2 times you will get 2 valid bitmaps back. Same for all the other similar tests here.